### PR TITLE
WD-3452 - Entity details tests

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add tests for entity details components.

## Details

https://warthogs.atlassian.net/browse/WD-3452

### CODE REVIEW

File: src/components/RelationIcon/RelationIcon.tsx [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
   }
   const application = applications[applicationName];
-  return (
-    <CharmIcon name={applicationName} charmId={application["charm-url"]} />
-  );
+  const charmId =
+    "charm-url" in application
+      ? application["charm-url"]
+      : application["charm"];
+  return <CharmIcon name={applicationName} charmId={charmId} />;
```
Comment (huwshimi) [inline src/components/RelationIcon/RelationIcon.tsx:20]: Drive-by fix so that icons appear.

File: src/pages/EntityDetails/EntityDetails.tsx [hunk 2]
```diff
@@ -78,17 +76,15 @@ const EntityDetails = () => {
   );
   const entityType = getEntityType(routeParams);
 
-  let credentials = null;
-  let controllerWSHost = "";
-  let wsProtocol: string | null = null;
-  if (primaryControllerData) {
-    credentials = getUserPass(storeState, primaryControllerData[0]);
-    controllerWSHost = primaryControllerData[0]
+  const credentials = useAppSelector((state) =>
```
Comment (huwshimi) [inline src/pages/EntityDetails/EntityDetails.tsx:79]: Just a bit of tidying up so that this uses a memoized selector.

File: src/pages/EntityDetails/Model/ApplicationsTab/ApplicationsTab.tsx [hunk 3]
```diff
@@ -56,7 +55,7 @@ const ContentRevealTitle = ({
     <span>
       {count} {pluralize(count, subject)}
     </span>
-    <ChipGroup chips={chips} descriptor={null} />
+    <ChipGroup chips={chips} className="u-no-margin" descriptor={null} />
```
Comment (huwshimi) [inline src/pages/EntityDetails/Model/ApplicationsTab/ApplicationsTab.tsx:58]: A drive-by spacing fix.

File: src/tables/tableRows.tsx [hunk 4]
```diff
@@ -603,20 +607,20 @@ export function generateOffersRows(modelStatusData: ModelData | null) {
...
           content: (
-            <TruncatedTooltip message={offer.applicationName}>
+            <TruncatedTooltip message={offer["application-name"]}>
               <RelationIcon
-                applicationName={offer.applicationName}
+                applicationName={offer["application-name"]}
                 applications={modelStatusData.applications}
               />
-              {offer.applicationName}
+              {offer["application-name"]}
```
Comment (huwshimi) [inline src/tables/tableRows.tsx:623]: This was displaying as blank because it was using the wrong casing here.

File: src/tables/tableRows.tsx [hunk 5]
```diff
@@ -626,7 +630,7 @@ export function generateOffersRows(modelStatusData: ModelData | null) {
           ),
         },
         {
-          content: offer.activeConnectedCount,
+          content: offer["active-connected-count"],
```
Comment (huwshimi) [inline src/tables/tableRows.tsx:633]: This was also blank.

File: src/tables/tableRows.tsx [hunk 6]
```diff
@@ -690,20 +694,20 @@ export function generateConsumedRows(modelStatusData?: ModelData | null) {
...
           content: (
-            <TruncatedTooltip message={application.offerName}>
+            <TruncatedTooltip message={application["offer-name"]}>
               <RelationIcon
-                applicationName={application.offerName}
+                applicationName={application["offer-name"]}
                 applications={modelStatusData.applications}
               />
-              {application.offerName}
+              {application["offer-name"]}
```
Comment (huwshimi) [inline src/tables/tableRows.tsx:710]: Another place that the data wasn't appearing due to the casing.

**Review by Lukewh — APPROVED** [review]
:+1: LGTM
